app/javascript/components/SignUp.js

```diff
@@ -45,7 +45,8 @@ function handleUserSignUp() {
         window.location.href = formatLink(link, userLocale);
       } else if (role === 'client' ) {
         link = '/search';
-        history.push(formatLink(link, userLocale), { signUp: true, currentUser, days });
+        const signupQuery = '?signup=true';
+        window.location.href = formatLink(link + signupQuery, userLocale);
```

Comment:
Great job! This looks really good and works great too. Only thing is to clean up any code that relies on the state from the previous history state object and make sure removing history state doesn't break anything. I believe there is code in the SearchBar.js that relies on the location state days as well as in NewAvailability.js. Otherwise. Looks good.

Reply:
@kscali Good idea. The location.state probably won't be used again. I've removed all references to the location.state in SearchBar and NewAvailability. Also, I searched the codebase, and it looks like NewAvailability is never called during the signup process, so I also removed any references to the signup state in NewAvailability.